projects/storefrontlib/shared/services/import-export/files-form-validators.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+import { AbstractControl, ValidationErrors, ValidatorFn } from '@angular/forms';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class FilesFormValidators {
+  /**
+   * Checks max size of file
+   *
+   * @param {number} maxSize Max size [MB]
+   * @returns Uses 'tooLarge' validator error with maxSize property
+   * @memberof FilesFormValidators
+   */
+  maxSize(maxSize?: number): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+      const errors: ValidationErrors = {};
+      if (maxSize) {
+        const files: File[] = Array.from(control.value);
+        files.forEach((file: File) => {
+          if (file.size > maxSize * 1000000) {
+            errors.tooLarge = { maxSize };
+          }
+        });
```

Comment:
nitpick: It's quite inconsistent:
- we validate many files at once
- but we expose only one error message, without saying which file exceeded the limit.

What about passing the `File` as the second parameter of the error object?

Reply:
I was thinking about it, but I was afraid of the huge structure of that object. But maybe we can push file names that are not valid into an array?
`tooLarge = { maxSize, files: [fileName1, filename2] }`